src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -134,11 +127,55 @@ class ddscxx_serdata : public ddsi_serdata {
     return t;
   }
 
-  T* getT()
-  {
+  T* getT() {
+    //TODO(Sumanth), clean this mess-up and make some functions
 #ifdef DDSCXX_HAS_SHM
     // if iox chunk is available, dont deserialize the sample, return the chunk directly
     if (iox_chunk != nullptr && data() == nullptr) {
+      auto iox_header = iceoryx_header_from_chunk(iox_chunk);
+
+      if (iox_header->shm_data_state == IOX_CHUNK_CONTAINS_SERIALIZED_DATA) {
+        T *t = m_t.load(std::memory_order_acquire);
+        if (t == nullptr) {
+          t = new T();
```

Comment:
I agree with all the above, but with the constraints here (i.e. how the constructors for the types are generated by the IDL compiler) I think we should be good here. If you agree I will propose to fix/update this in a follow-up MR.

Reply:
Yes, it depends on what the deserialization expects. A constructed object or raw memory. I think in this case it is the former and so we even need to construct `T`. Note that ideally I would prefer to leave this to the deserialization (and just give it memory) because it can optimize it better (since the deserialization code alone knows what has to be done).